File: webapp/src/types/PagedResponse.ts
/**
 * Generic paged response wrapper for API responses
 * Converted from Java PagedResponse model
 */
export interface PagedResponse<T> {
  content: T[];
  totalElements: number;
  totalPages: number;
  size: number;
  number: number; // Current page number (0-based)
  numberOfElements: number;
  first: boolean;
  last: boolean;
  empty: boolean;
  pageable?: unknown;
  sort?: unknown;
}

/**
 * Pagination parameters for API requests
 */
export interface PaginationParams {
  page?: number; // 0-based page number
  size?: number; // Page size
  sort?: string; // Sort field and direction (e.g., "name,asc")
}

/**
 * Utility functions for PagedResponse operations
 */
export const PagedResponseUtils = {
  /**
   * Check if response has content
   */
  hasContent: <T>(response: PagedResponse<T>): boolean => {
    return response.content && response.content.length > 0;
  },

  /**
   * Check if there's a next page
   */
  hasNext: <T>(response: PagedResponse<T>): boolean => {
    return !response.last;
  },

  /**
   * Check if there's a previous page
   */
  hasPrevious: <T>(response: PagedResponse<T>): boolean => {
    return !response.first;
  },

  /**
   * Get next page number (1-based for UI display)
   */
  getNextPageNumber: <T>(response: PagedResponse<T>): number | null => {
    return PagedResponseUtils.hasNext(response) ? response.number + 2 : null;
  },

  /**
   * Get previous page number (1-based for UI display)
   */
  getPreviousPageNumber: <T>(response: PagedResponse<T>): number | null => {
    return PagedResponseUtils.hasPrevious(response) ? response.number : null;
  },

  /**
   * Get current page number (1-based for UI display)
   */
  getCurrentPageNumber: <T>(response: PagedResponse<T>): number => {
    return response.number + 1;
  },

  /**
   * Get page range text (e.g., "Showing 1-10 of 25")
   */
  getPageRangeText: <T>(response: PagedResponse<T>): string => {
    if (response.empty) {
      return 'No items found';
    }

    const start = response.number * response.size + 1;
    const end = Math.min(start + response.numberOfElements - 1, response.totalElements);
    
    return `Showing ${start}-${end} of ${response.totalElements}`;
  },

  /**
   * Create default pagination params
   */
  createDefaultParams: (overrides?: Partial<PaginationParams>): PaginationParams => {
    return {
      page: 0,
      size: 20,
      sort: 'id,desc',
      ...overrides
    };
  }
};

File: .cli_backup/src/main/java/com/localtechsupport/cli/model/PagedResponse.java
package com.localtechsupport.cli.model;

import com.fasterxml.jackson.annotation.JsonProperty;
import java.util.List;

/**
 * Generic paged response wrapper for API responses
 * 
 * @param <T> The type of content in the response
 */
public class PagedResponse<T> {

    @JsonProperty("content")
    private List<T> content;

    @JsonProperty("totalElements")
    private Long totalElements;

    @JsonProperty("totalPages")
    private Integer totalPages;

    @JsonProperty("size")
    private Integer size;

    @JsonProperty("number")
    private Integer number;

    @JsonProperty("numberOfElements")
    private Integer numberOfElements;

    @JsonProperty("first")
    private Boolean first;

    @JsonProperty("last")
    private Boolean last;

    @JsonProperty("empty")
    private Boolean empty;

    @JsonProperty("pageable")
    private Object pageable;

    @JsonProperty("sort")
    private Object sort;

    // Default constructor
    public PagedResponse() {}

    // Getters and Setters
    public List<T> getContent() {
        return content;
    }

    public void setContent(List<T> content) {
        this.content = content;
    }

    public Long getTotalElements() {
        return totalElements;
    }

    public void setTotalElements(Long totalElements) {
        this.totalElements = totalElements;
    }

    public Integer getTotalPages() {
        return totalPages;
    }

    public void setTotalPages(Integer totalPages) {
        this.totalPages = totalPages;
    }

    public Integer getSize() {
        return size;
    }

    public void setSize(Integer size) {
        this.size = size;
    }

    public Integer getNumber() {
        return number;
    }

    public void setNumber(Integer number) {
        this.number = number;
    }

    public Integer getNumberOfElements() {
        return numberOfElements;
    }

    public void setNumberOfElements(Integer numberOfElements) {
        this.numberOfElements = numberOfElements;
    }

    public Boolean isFirst() {
        return first;
    }

    public void setFirst(Boolean first) {
        this.first = first;
    }

    public Boolean isLast() {
        return last;
    }

    public void setLast(Boolean last) {
        this.last = last;
    }

    public Boolean isEmpty() {
        return empty;
    }

    public void setEmpty(Boolean empty) {
        this.empty = empty;
    }

    // Utility methods
    public boolean hasContent() {
        return content != null && !content.isEmpty();
    }

    public boolean hasNext() {
        return !Boolean.TRUE.equals(last);
    }

    public boolean hasPrevious() {
        return !Boolean.TRUE.equals(first);
    }

    @Override
    public String toString() {
        return String.format("PagedResponse{totalElements=%d, totalPages=%d, size=%d, number=%d}", 
                           totalElements, totalPages, size, number);
    }
} 